### PR TITLE
Bugfix Library

### PR DESCRIPTION
**Pull request by nyamsprod**
- bug fix stream filter #72
- improve public API by deprecating (get|set)EncodingFrom

### CODE REVIEW

File: src/AbstractCsv.php [hunk 1]
```diff
@@ -190,7 +190,7 @@ protected function newInstance($class, $open_mode)
         $csv->delimiter = $this->delimiter;
         $csv->enclosure = $this->enclosure;
         $csv->escape = $this->escape;
-        $csv->encodingFrom = $this->encodingFrom;
+        $csv->input_encoding = $this->input_encoding;
```
Comment (hannesvdvreken) [inline src/AbstractCsv.php]: just curious: why snake_case instead of camelCase?

Comment (nyamsprod) [inline src/AbstractCsv.php]: I was in the mood for snake case :) . Seriously I did this to match $input_bom and $output_bom which are already in snake case. I should probably review all the codebase and make a clear decision between both coding style. Of note none of them are in contradiction to PSR2 if I recall but yes the code needs consistency.

Comment (hannesvdvreken) [inline src/AbstractCsv.php]: > I was in the mood for snake case
:smile: 

> none of them are in contradiction to PSR2
You're right

File: src/Config/Output.php [hunk 2]
```diff
@@ -54,26 +54,56 @@
...
 
         return $this;
     }
 
     /**
+     * Sets the CSV encoding charset
+     *
+     * DEPRECATION WARNING! This method will be removed in the next major point release
+     *
+     * @deprecated deprecated since version 4.1
```
Comment (vlakoff) [inline src/Config/Output.php]: since version 4.1? should be 8.x.x (to be released)

File: src/Config/Output.php [hunk 3]
```diff
@@ -54,26 +54,56 @@
...
+    {
+        return $this->input_encoding;
+    }
+
+    /**
      * Gets the source CSV encoding charset
      *
+     * DEPRECATION WARNING! This method will be removed in the next major point release
+     *
+     * @deprecated deprecated since version 4.1
```
Comment (vlakoff) [inline src/Config/Output.php]: same here

File: src/Config/Output.php (hunk 2 above)
Comment (hannesvdvreken) [inline src/Config/Output.php]: It's deprecated sinde 4.1, and will be removed in the next major version.

Comment (vlakoff) [inline src/Config/Output.php]: I just thought of some typo / old PR. Okay then, I trust you :)

**Comment by nyamsprod** [comment]
@vlakoff indeed you are correct it is a typo and the deprecation message was changed in the master. :+1: 

I should point out that I had a hard time finding this discussion since the comment was made on an already closed PR

**Pull request merged**